Automattic/woocommerce-payments
# Add an endpoint to register a terminal card reader

Fixes #3432

This PR aims to add a new endpoint to allow the registration of a card reader against a given terminal location. Usually, explicit server-side card registration is not needed. In the case of Bluetooth readers, this is internally handled by the Stripe Mobile SDK.

In the case of Wifi-only readers (eg. the counter-top variety), however, it isn't possible for the mobile app to do so since it is not directly connected to the reader. The connection between mobile app and reader is via Stripe.

#### Testing instructions

Pre-requisites:

* Registering a reader requires a terminal location. A new location may be created via `POST /payments/terminal/locations`. See #3013.
* A real Wifi card reader, such as BBPOS WisePOS E, to get a registration code. Simulated readers won't provide registration codes.

Instructions:

* Make a POST request to `/payments/terminal/readers` with the required params `{ location: tml_your_loc_id, registration_code: your_code }`.
* Verify the response returned by the API call is a 200 response and contains the new reader object.
* Not providing required params `location` and/or `registration_code` should return a 400 response.

-------------------

- [ ] Added changelog entries to both `readme.txt` and `changelog.txt` (does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (does not apply)
